ISOLATE_STDLIB_MODULES.rst

```diff
@@ -0,0 +1,240 @@
+PEP: XXX
+Title: Isolating modules in the standard library
+Author: Petr Viktorin <encukou@gmail.com>, Erlend Egeberg Aasland <erlend.aasland@protonmail.com>
+Discussions-To: XXX
+Status: Draft
+Type: Standards
+Requires: 489, 573, 630
+Created: 16-Feb-2022
+Python-Version: 3.11
+Post-History: XXX
+
+
+Abstract
+========
+
+Extensions in the standard library will be converted to multi-phase
+initialization (:pep:`489`) and where possible, all state will be
+stored on module objects rather than in process-global variables.
+
+
+Note on Backdating
+==================
+
+Much of this proposal has already been implemented, partly to solve related
+issues and partly because of misunderstandings and lack of coordination.
+
+We submit this PEP to explain the changes, seek consensus on
+whether they are good,  propose changing modules that do not exhibit
+issues, and set best practices for new modules.
+
+We are aware that the pre-existing changes might need to be reverted if issues
+are found.
```

Comment:
The first heap types were added in Python 3.2. Some of these changes are _very_ old. If one concludes there is no gain in converting new types (net win = 0), then there is also no gain in reverting very old changes; IMO it would just be needless code churn, which would be very counter productive (risk of introducing new regressions, require a _lot_ of reviewer resources).

There is already a section further down that says if issues are found, they must be resolved. That should suffice. I suggest dropping this sentence.

Reply:
> There is already a section further down that says if issues are found, they must be resolved.

Issues could be general (the overall process) or specific (for example module or type specific). One way of resolving a module specific issue could be to revert changes for that specific module; that can be resolved on a bug ticket; we don't need it in the PEP.